src/ansi-c/library/stdio.c

```diff
@@ -232,18 +232,21 @@ char *fgets(char *str, int size, FILE *stream)
   __CPROVER_bool error=__VERIFIER_nondet___CPROVER_bool();
 
   (void)size;
-  #if !defined(__linux__) || defined(__GLIBC__)
-  (void)*stream;
-  #else
-  (void)*(char*)stream;
-  #endif
+  if(stream != stdin)
+  {
+#if !defined(__linux__) || defined(__GLIBC__)
+    (void)*stream;
+#else
+    (void)*(char *)stream;
+#endif
```

Comment:
I think the VCCs get a bit easier if the type matches that of the allocation.

Reply:
Yes, but that (`__linux__ && !__GLIBC__`) is the case where `FILE` remains an incomplete type.